src/app/loadout/loadout.service.ts

```diff
@@ -179,10 +191,34 @@ function LoadoutService(): LoadoutServiceType {
     await saveLoadouts(reduxStore.getState().loadouts.loadouts);
   }
 
-  async function saveLoadout(loadout: Loadout): Promise<void> {
-    await getLoadouts(); // make sure we have loaded all loadouts first!
-    reduxStore.dispatch(actions.updateLoadout(loadout));
-    await saveLoadouts(reduxStore.getState().loadouts.loadouts);
+  async function saveLoadout(loadout: Loadout): Promise<Loadout | undefined> {
+    const loadouts = await getLoadouts();
+    const clashingLoadout = getClashingLoadout(loadouts, loadout);
+
+    if (!clashingLoadout) {
+      reduxStore.dispatch(actions.updateLoadout(loadout));
+      await saveLoadouts(reduxStore.getState().loadouts.loadouts);
+    }
+
+    return clashingLoadout;
+  }
+
+  function getClashingLoadout(loadouts: Loadout[], newLoadout: Loadout): Loadout | undefined {
+    return _.find(loadouts, (loadout) => {
+      return doLoadoutsClash(loadout, newLoadout);
+    });
+  }
+
+  function doLoadoutsClash(loadout1: Loadout, loadout2: Loadout): boolean {
+    if (loadout1.name !== loadout2.name) {
+      return false;
+    }
+
+    if (loadout1.id === loadout2.id) {
+      return false;
+    }
+
+    return loadout1.classType === loadout2.classType;
```

Comment:
Like I mentioned earlier, this is a touch verbose. I'd have written the whole thing out something like:

```ts
return loadouts.find((loadout) => 
  loadout.id !== newLoadout.id &&
  loadout.name === newLoadout.name &&
  loadout.classType === newLoadout.classType);
```

Reply:
Done, the only time I prefer the more verbose kind is when there are multiple conditions. Just removes some of the mental load if someone has to know what it is doing.